components/ai_chat/core/browser/engine/oai_api_client.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_ENGINE_OAI_API_CLIENT_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CORE_BROWSER_ENGINE_OAI_API_CLIENT_H_
+
+#include <string>
+#include <vector>
+
+#include "brave/components/ai_chat/core/browser/ai_chat_credential_manager.h"
+#include "brave/components/ai_chat/core/browser/engine/engine_consumer.h"
+#include "brave/components/ai_chat/core/common/mojom/ai_chat.mojom-forward.h"
+
+namespace api_request_helper {
+class APIRequestResult;
+}  // namespace api_request_helper
+
+namespace network {
+class SharedURLLoaderFactory;
+}  // namespace network
+
+namespace ai_chat {
+
+// Performs remote request to the OAI format APIs.
+class OAIAPIClient {
```

Comment:
This class needs unit tests, especially around its handling of responses.

Reply:
added